Automattic/wp-calypso
# ImporterSignup: Validate input

> Note: This PR supersedes #26833

This PR adds some basic validation to our URL input field found in the Importer Nux flow.
Currently, we check only that the URL is either valid or invalid, showing a generic, non-nuanced validation message when an invalid value is typed.

For now, I've pulled out the `fetchIsSiteImportable` logic, which was removed. This behaviour will be re-implemented soon in the next step of this signup flow.

### Screenshots

#### Invalid input
The following screenshots show each of the 3 states to be found in this work and the copy currently found here:
<img width="493" alt="screen shot 2018-09-16 at 15 32 49" src="https://user-images.githubusercontent.com/4335450/45597553-faaae600-b9c5-11e8-8ed4-adbf323bb98f.png">

#### Neutral messaging
<img width="493" alt="screen shot 2018-09-16 at 15 33 01" src="https://user-images.githubusercontent.com/4335450/45597552-faaae600-b9c5-11e8-984f-285f3ad85abd.png">

#### Valid input (after blur)
<img width="493" alt="screen shot 2018-09-16 at 15 33 12" src="https://user-images.githubusercontent.com/4335450/45597551-faaae600-b9c5-11e8-8bc1-8524cb181383.png">

### To Test

- Run this branch locally
- Visit http://calypso.localhost:3000/start/from-site
- It should redirect to http://calypso.localhost:3000/start/from-site/import-from-url & you should see a new step with a form to enter your URL.
- Start typing something invalid (no `.`s for example)
- Try to submit the form field
  - You should see a validation message appear in place of the default text
- Start typing again
  - You should see the error messaging change back to the default text
- Make sure the input is a valid url and press <kbd>tab</kbd>
  - You should now see copy showing that the address is valid
- Hit the continue button
  - You should go to the next step successfully
